server/resp/src/main/java/org/infinispan/server/resp/commands/cluster/SegmentSlotRelation.java

```diff
@@ -0,0 +1,53 @@
+package org.infinispan.server.resp.commands.cluster;
+
+import static org.infinispan.commons.logging.Log.CONFIG;
+
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.commons.util.IntSets;
+import org.infinispan.commons.util.Util;
+
+public class SegmentSlotRelation {
+
+   // This is the default number of slots a clustered Redis has. Which is not configurable.
+   // See: https://redis.io/docs/reference/cluster-spec/#key-distribution-model
+   public static final int SLOT_SIZE = 1 << 14;
+   private final int segmentCount;
+   private final int width;
+
+   public SegmentSlotRelation(int segmentSize) {
+      if (!Util.isPow2(segmentSize))
+         throw CONFIG.respCacheSegmentSizePow2(segmentSize);
+      this.segmentCount = segmentSize;
+      if (segmentSize > SLOT_SIZE) {
+         throw new IllegalArgumentException("Number of segments cannot be larger than " + SLOT_SIZE);
+      }
+      this.width = SLOT_SIZE / segmentSize;
+   }
+
+   public int hashToSlot(int hash) {
+      return hash % SLOT_SIZE;
+   }
+
+   public IntSet segmentToSlots(int segment) {
+      assert segment >= 0 && segment < segmentCount : "Not provided a segment value";
+
+      // TODO: this code can be optimized as a new IntSet that offsets an existing IntSet
```

Comment:
I would like to  do this. Although we technically don't need the new IntSets for this use case if we just use the slotWidth in the calling code.

Reply:
I added an implementation. If you think it is too much we can remove it and do everything on the caller. There is only one place left using this method.